src/main/java/com/gdschongik/gdsc/domain/integration/JavaMailSender.java

```diff
@@ -0,0 +1,65 @@
+package com.gdschongik.gdsc.domain.integration;
+
+import static com.gdschongik.gdsc.global.common.constant.EmailConstant.SENDER_ADDRESS;
+import static com.gdschongik.gdsc.global.common.constant.EmailConstant.SENDER_PERSONAL;
+
+import com.gdschongik.gdsc.global.exception.CustomException;
+import com.gdschongik.gdsc.global.exception.ErrorCode;
+import jakarta.mail.Message.RecipientType;
+import jakarta.mail.MessagingException;
+import jakarta.mail.internet.InternetAddress;
+import jakarta.mail.internet.MimeMessage;
+import java.io.UnsupportedEncodingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class JavaMailSender implements MailSender {
+
+    private final org.springframework.mail.javamail.JavaMailSender javaMailSender;
+
+    @Override
+    public void send(String recipient, String subject, String content) {
+        MimeMessage message = writeMimeMessage(recipient, subject, content);
+        javaMailSender.send(message);
+    }
+
+    private MimeMessage writeMimeMessage(String recipient, String subject, String content) {
+        MimeMessage message = javaMailSender.createMimeMessage();
+        addRecipients(message, recipient);
+        setMessageAttributes(message, subject, content);
+        return message;
+    }
+
+    private void addRecipients(MimeMessage message, String recipient) {
+        try {
+            message.addRecipients(RecipientType.TO, recipient);
+        } catch (MessagingException e) {
+            throwCustomExceptionWithAdditionalMessage(ErrorCode.MESSAGING_EXCEPTION, e.getMessage());
```

Comment:
바로 new CustomException으로 던져도 되지 않나요?
별도 메서드로 추출하신 이유가 궁금합니다.
```suggestion
            throw new CustomException(ErrorCode.MESSAGING_EXCEPTION, e.getMessage());
```

Reply:
일단 위 try-catch문에선 MessagingException를 잡고 있습니다. 이 MessagingException는 상황에 따라 message를 통해 exception 발생 사유를 간단하게 설명합니다. 

개인적으로 느끼기엔 간단한 설명이 부족하게 느껴져서 MESSAGING_EXCEPTION이 가지고 있는 메시지인 "수신자 이메일이 올바르지 않습니다."와 MessagingException이 건내주는 message를 합치고 싶었습니다.

예를 들어 MessagingException이 전달하는 메시지가 "No From address"인 경우, "수신자 이메일이 올바르지 않습니다. : No From address"와 같은 형태로 합쳐서 건내주고자 했습니다.

이런 상황에선 아래와 같은 코드가 catch문에 들어가게 되었는데요?

```java
String errorMessage = ErrorCode.MESSAGING_EXCEPTION.getMessage() + " : " + e.getMessage();
throw new CustomException(ErrorCode.MESSAGING_EXCEPTION, errorMessage);
```

이 2줄의 코드가 두 군데에서 중복되어 사용되기도 하고,
읽는 사람에게 추가적인 메시지를 더하겠다는 의도를 전달하고 싶어 메서드로 분리하게 되었습니다~